modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/validation/OperationParametersModelValidator.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.introspection.validation;
+
+import static java.lang.String.format;
+import static org.mule.runtime.module.extension.internal.ExtensionProperties.TARGET_ATTRIBUTE;
+import static org.mule.runtime.module.extension.internal.ExtensionProperties.TRANSACTIONAL_ACTION_PARAMETER_NAME;
+import org.mule.runtime.extension.api.exception.IllegalModelDefinitionException;
+import org.mule.runtime.extension.api.introspection.ExtensionModel;
+import org.mule.runtime.extension.api.introspection.operation.OperationModel;
+import org.mule.runtime.extension.api.introspection.parameter.ParameterModel;
+import org.mule.runtime.module.extension.internal.exception.IllegalOperationModelDefinitionException;
+import org.mule.runtime.module.extension.internal.util.IdempotentExtensionWalker;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.LinkedHashMultimap;
+import com.google.common.collect.Multimap;
+
+import java.util.List;
+
+/**
+ * Validates that no {@link ParameterModel parameters} named {@code target}, since that word is reserved.
+ *
+ * @since 4.0
+ */
+public final class OperationParametersModelValidator implements ModelValidator
+{
+
+    private final List<String> reservedWords = ImmutableList.of(TARGET_ATTRIBUTE, TRANSACTIONAL_ACTION_PARAMETER_NAME);
```

Comment:
Should not the test just check that every word in this list is detected instead of having a different test for each one?